clinicadl/utils/caps_dataset/caps_dataset_refactoring/caps_dataset.py

```diff
@@ -0,0 +1,1224 @@
+        Returns:
+            dictionary with following items:
+                - "image" (torch.Tensor): the input given to the model,
+                - "label" (int or float): the label used in criterion,
+                - "participant_id" (str): ID of the participant,
+                - "session_id" (str): ID of the session,
+                - f"{self.mode}_id" (int): number of the element,
+                - "image_path": path to the image loaded in CAPS.
+
+        """
+        pass
+
+    @abc.abstractmethod
+    def num_elem_per_image(self) -> int:
+        """Computes the number of elements per image based on the full image."""
+        pass
+
+    def size_elem(self):
+        """Computes the number of elements per image based on the full image."""
+        return self[0]["image"].size()
+
+    def eval(self):
+        """Put the dataset on evaluation mode (data augmentation is not performed)."""
+        self.eval_mode = True
+        return self
+
+    def train(self):
+        """Put the dataset on training mode (data augmentation is performed)."""
+        self.eval_mode = False
+        return self
+
+
+class CapsDatasetImage(CapsDataset):
+    """Dataset of MRI organized in a CAPS folder."""
+
+    def __init__(
+        self,
+        caps_directory: Path,
+        tsv_label: Path,
+        preprocessing_dict: Dict[str, Any],
+        train_transformations: Optional[Callable] = None,
+        label_presence: bool = True,
+        label: str = None,
+        label_code: Dict[str, int] = None,
+        all_transformations: Optional[Callable] = None,
+    ):
+        """
+        Args:
+            caps_directory: Directory of all the images.
+            data_file: Path to the tsv file or DataFrame containing the subject/session list.
+            preprocessing_dict: preprocessing dict contained in the JSON file of prepare_data.
+            train_transformations: Optional transform to be applied only on training mode.
+            label_presence: If True the diagnosis will be extracted from the given DataFrame.
+            label: Name of the column in data_df containing the label.
+            label_code: label code that links the output node number to label value.
+            all_transformations: Optional transform to be applied during training and evaluation.
+            multi_cohort: If True caps_directory is the path to a TSV file linking cohort names and paths.
+
+        """
+
+        self.mode = "image"
+        super().__init__(
+            caps_directory,
+            tsv_label,
+            preprocessing_dict,
+            augmentation_transformations=train_transformations,
+            label_presence=label_presence,
+            label=label,
+            label_code=label_code,
+            transformations=all_transformations,
+        )
+
+        self.prepare_dl = self.preprocessing_dict["prepare_dl"]
+
+    @property
+    def elem_index(self):
+        return None
+
+    def __getitem__(self, idx):
```

Comment:
Returns a `Sample` or a `Dict[str, Any]` depending on what you decide.

```suggestion
    def __getitem__(self, idx) -> Sample:
```